scipy/stats/tests/test_multivariate.py

```diff
@@ -1635,6 +1643,248 @@ def test_haar(self):
+        dist = multivariate_t()
+        assert_equal(dist.loc, [0])
+        assert_equal(dist.shape, [[1]])
+        assert (dist.df == 1)
+
+    def test_argument_handling(self):
+        dist = multivariate_t(loc=None, shape=None, df=None)
+        assert_equal(dist.loc, 0)
+        assert_equal(dist.shape, 1)
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=None, shape=None, df=7)
+        assert_equal(dist.loc, 0)
+        assert_equal(dist.shape, 1)
+        assert (dist.df == 7)
+
+        dist = multivariate_t(loc=None, shape=[[7, 0], [0, 7]], df=None)
+        assert_equal(dist.loc, [0, 0])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=None, shape=[[7, 0], [0, 7]], df=7)
+        assert_equal(dist.loc, [0, 0])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 7)
+
+        dist = multivariate_t(loc=[7, 7], shape=None, df=None)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[1, 0], [0, 1]])
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=[7, 7], shape=None, df=7)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[1, 0], [0, 1]])
+        assert (dist.df == 7)
+
+        dist = multivariate_t(loc=[7, 7], shape=[[7, 0], [0, 7]], df=None)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 1)
+
+        dist = multivariate_t(loc=[7, 7], shape=[[7, 0], [0, 7]], df=7)
+        assert_equal(dist.loc, [7, 7])
+        assert_equal(dist.shape, [[7, 0], [0, 7]])
+        assert (dist.df == 7)
+
+    def test_argument_error_handling(self):
+        # `loc` should be a one-dimensional vector.
+        with pytest.raises(ValueError):
+            multivariate_t(loc=[[1, 1]])
+
+        # `shape` should be scalar or square matrix.
+        with pytest.raises(ValueError):
+            multivariate_t(shape=[[1, 1], [2, 2], [3, 3]])
+
+        # Confirm df > 0, otherwise `ValueError`:
+        with pytest.raises(ValueError):
+            multivariate_t(df=-1)
+        with pytest.raises(ValueError):
+            multivariate_t(df=0)
+
+        # Floats are okay:
+        try:
+            multivariate_t(df=5.0)
+        except ValueError:
+            pytest.fail("Test failure: `df` should allow floats.")
+
+    def test_reproducibility(self):
+        rng = np.random.RandomState(4)
+        loc = rng.uniform(size=3)
+        shape = np.eye(3)
+        dist1 = multivariate_t(loc, shape, df=3, seed=2)
+        dist2 = multivariate_t(loc, shape, df=3, seed=2)
+        samples1 = dist1.rvs(size=10)
+        samples2 = dist2.rvs(size=10)
+        assert_equal(samples1, samples2)
+
+    def test_scalar_list_and_ndarray_arguments(self):
+        dist = multivariate_t(-1, 2, 3)
+        assert(dist.loc == np.array([-1]))
+        assert(dist.shape == np.array([2]))
+        assert(dist.df == 3)
```

Comment:
suggest to swap `assert` with `assert_{allclose,equal}` here and throughout for consistency.

Reply:
Okay, I'm now using `assert_equal`. This catches some small discrepancies. For example, this passes an `assert`:

```python
dist.shape == np.array([2])
# array([[ True]])
```

but not an `assert_equal`. The real value should be `[[2]]` or `np.array([[2]])`.

I've also wrapped the test using `parameterize`.

Reply:
weird, must be a consistency issue w/the github UI, I now see changes, marking resolved, apologies for noise.